src/MyMarketManager.Data/Services/ImageVectorizationProcessor.cs

```diff
@@ -0,0 +1,252 @@
+                {
+                    results.Add(new ProductImageSearchResult
+                    {
+                        ProductPhoto = photo,
+                        Product = photo.Product,
+                        SimilarityScore = similarity
+                    });
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogWarning(ex, "Error calculating similarity for photo {PhotoId}", photo.Id);
+            }
+        }
+
+        // Return top N results ordered by similarity
+        return results
+            .OrderByDescending(r => r.SimilarityScore)
+            .Take(maxResults)
+            .ToList();
+    }
+
+    /// <summary>
+    /// Searches for product images using semantic text search.
+    /// Note: For large datasets, consider implementing pagination or using a vector database.
+    /// </summary>
+    public async Task<List<ProductImageSearchResult>> SearchByTextAsync(
+        string searchText,
+        int maxResults = 10,
+        float similarityThreshold = 0.6f,
+        CancellationToken cancellationToken = default)
+    {
+        // Vectorize the search text
+        var searchVector = await _visionService.VectorizeTextAsync(searchText, cancellationToken);
+
+        // Get vectorized images (for large datasets, consider pagination or filtering)
+        var allPhotos = await _context.ProductPhotos
+            .Include(p => p.Product)
+            .Where(p => p.VectorEmbedding != null)
+            .Take(1000) // Limit to prevent memory issues with large datasets
+            .ToListAsync(cancellationToken);
+
+        // Calculate similarity scores
+        var results = new List<ProductImageSearchResult>();
+
+        foreach (var photo in allPhotos)
+        {
+            try
+            {
+                var photoVector = JsonSerializer.Deserialize<float[]>(photo.VectorEmbedding!);
+                if (photoVector == null) continue;
+
+                var similarity = _visionService.CalculateCosineSimilarity(searchVector, photoVector);
+
+                if (similarity >= similarityThreshold)
+                {
+                    results.Add(new ProductImageSearchResult
+                    {
+                        ProductPhoto = photo,
+                        Product = photo.Product,
+                        SimilarityScore = similarity
+                    });
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogWarning(ex, "Error calculating similarity for photo {PhotoId}", photo.Id);
+            }
+        }
+
+        // Return top N results ordered by similarity
+        return results
+            .OrderByDescending(r => r.SimilarityScore)
+            .Take(maxResults)
+            .ToList();
+    }
+
+    /// <summary>
+    /// Searches for products by tags.
+    /// </summary>
+    public async Task<List<ProductImageSearchResult>> SearchByTagsAsync(
```

Comment:
I think we should simplify and just drop the whole tagging feature for now.

Focus on the text and image search.

Clean up the entities too.